agent/consul/session_timers.go

```diff
@@ -0,0 +1,82 @@
+package consul
+
+import (
+	"sync"
+	"time"
+)
+
+// SessionTimers provides a map of named timers which
+// is safe for concurrent use.
+type SessionTimers struct {
+	sync.RWMutex
+	m map[string]*time.Timer
+}
+
+func NewSessionTimers() *SessionTimers {
+	return &SessionTimers{m: make(map[string]*time.Timer)}
+}
+
+// Get returns the timer with the given id or nil.
+func (t *SessionTimers) Get(id string) *time.Timer {
+	t.RLock()
+	defer t.RUnlock()
+	return t.m[id]
+}
+
+// Set stores the timer under given id. If tm is nil the timer
+// witht the given id is removed.
+func (t *SessionTimers) Set(id string, tm *time.Timer) {
+	t.Lock()
+	defer t.Unlock()
+	if tm == nil {
```

Comment:
I actually don't mind `var, ok := t.Get("foo")` for the `Get()` side, since that's easier to remember to check and less likely for a `nil` to go on somewhere else.